### PR TITLE
Change variant decorator to tag

### PR DESCRIPTION
**Pull request by Vcagnion**
Use tag instead of variant that contain "variant" and "branch_level" for more consistency and a better readability.
- Tested and Unit Tested
- .rst updated
- Updated all decorators

### CODE REVIEW

**Comment by sebastianpfischer** [comment]
Please do not use internal ticket system in the branch you are using.
Also rename the commit to: `refactor!: <title>`

**Review by sebastianpfischer — APPROVED** [review]

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]
Please replace the commit name

**Review by BKaDamien — APPROVED** [review]

File: docs/introduction/integration_test.rst [hunk 1]
```diff
@@ -419,7 +419,7 @@ Find below a full example for a test suite/case declaration :
   -> run_timeout : time to wait for a report 10 seconds during test_run
   -> teardown_timeout : time to wait for a report 3 seconds during teardown
   -> test_ids: [optional] store the requirements into the report
-  -> variant: [optional] list of variances if a subset of tests need to be executed
+  -> tag: [optional] list of variances if a subset of tests need to be executed
```
Comment (sebclrsn) [inline docs/introduction/integration_test.rst:422]:
```suggestion
  -> tag: [optional] dictionary containing lists of variants and/or test levels when only a subset of tests needs to be executed
```

File: src/pykiso/test_coordinator/test_case.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def __init__(
             wait for a report during teardown execution
         :param test_ids: jama references to get the coverage
             eg: {"Component1": ["Req1", "Req2"], "Component2": ["Req3"]}
-        :param variant: string that allows the user to execute a subset of tests
+        :param tag: dict that allows the user to execute a subset of tests
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_case.py:143]:
```suggestion
        :param tag: dictionary containing lists of variants and/or test levels when only a subset of tests needs to be executed
```

File: src/pykiso/test_coordinator/test_case.py [hunk 3]
```diff
@@ -213,7 +213,7 @@ def define_test_parameters(
     run_timeout: Optional[int] = None,
     teardown_timeout: Optional[int] = None,
     test_ids: Optional[dict] = None,
-    variant: Optional[dict] = None,
+    tag: Optional[dict] = None,
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_case.py:216]: I'd really go for more complete type hints
```suggestion
    tag: Optional[Dict[str, List[str]]] = None,
```

File: src/pykiso/test_coordinator/test_suite.py [hunk 4]
```diff
@@ -52,7 +52,7 @@ def __init__(
         run_timeout: Union[int, None],
         teardown_timeout: Union[int, None],
         test_ids: Union[dict, None],
-        variant: Union[list, None],
+        tag: Union[dict, None],
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_suite.py:55]: Same here for the type-hint

File: src/pykiso/test_coordinator/test_suite.py [hunk 5]
```diff
@@ -128,7 +128,7 @@ def __init__(
             wait for a report during teardown execution
         :param test_ids: jama references to get the coverage
             eg: {"Component1": ["Req1", "Req2"], "Component2": ["Req3"]}
-        :param variant: string that allows the user to execute a subset of tests
+        :param tag: string that allows the user to execute a subset of tests
```
Comment (sebclrsn) [inline src/pykiso/test_coordinator/test_suite.py:131]: Same in all of this file regarding the parameter documentation

**Review by sebclrsn — CHANGES_REQUESTED** [review]

**Review by sebclrsn — APPROVED** [review]

File: src/pykiso/test_coordinator/test_suite.py (hunk 5 above)
Comment (Vcagnion) [reply src/pykiso/test_coordinator/test_suite.py]: Done

File: src/pykiso/test_coordinator/test_suite.py (hunk 4 above)
Comment (Vcagnion) [reply src/pykiso/test_coordinator/test_suite.py]: Done

File: src/pykiso/test_coordinator/test_case.py (hunk 3 above)
Comment (Vcagnion) [reply src/pykiso/test_coordinator/test_case.py]: Done

File: src/pykiso/test_coordinator/test_case.py (hunk 2 above)
Comment (Vcagnion) [reply src/pykiso/test_coordinator/test_case.py]: Done

File: docs/introduction/integration_test.rst (hunk 1 above)
Comment (Vcagnion) [reply docs/introduction/integration_test.rst]: Done

**Review by kakooloukia — APPROVED** [review]

**Comment by kakooloukia** [comment]
@sebastianpfischer , do we cancel that PR due to branch name and re-create it from a new one ?

- [Change branch name]

**Comment by sebastianpfischer** [comment]
It may be difficult @kakooloukia. Let us just have the correct commit message instead

**Review by sebastianpfischer — APPROVED** [review]

**Pull request merged**